test/framework/deployment_helpers.go

```diff
@@ -241,7 +241,11 @@ func WaitForDNSUpgrade(ctx context.Context, input WaitForDNSUpgradeInput, interv
 		if err := input.Getter.Get(ctx, client.ObjectKey{Name: "coredns", Namespace: metav1.NamespaceSystem}, d); err != nil {
 			return false, err
 		}
-		if d.Spec.Template.Spec.Containers[0].Image == "k8s.gcr.io/coredns:"+input.DNSVersion {
+		// coredns image name has changed. So for now we accept:
+		// k8s.gcr.io/coredns:1.7.0
+		// k8s.gcr.io/coredns/coredns:v1.7.0
+		// k8s.gcr.io/coredns/coredns:v1.8.0
+		if strings.HasSuffix(d.Spec.Template.Spec.Containers[0].Image, "coredns:"+input.DNSVersion) {
```

Comment:
nit: given that we are concerned concerned here is the version, let's check for ":" version suffix only.
Also what about changing the comment into
```suggestion
		// NOTE: coredns image name has changed over time (k8s.gcr.io/coredns, 
		// k8s.gcr.io/coredns/coredns), so we are checking only the version actually changed.
		if strings.HasSuffix(d.Spec.Template.Spec.Containers[0].Image, input.DNSVersion) {
```

Reply:
Looking at this you'd still want to have a colon prefix to the check to make sure you're absolutely getting the right version

Eg if we don't then version `11.7.0` could be allowed as well right?

Reply:
Sounds good to me.